pkg/apis/v1/ec2nodeclass.go

```diff
@@ -487,9 +489,26 @@ type EC2NodeClass struct {
 // 3. A field is removed from the hash calculations
 const EC2NodeClassHashVersion = "v4"
 
-func (in *EC2NodeClass) Hash() string {
+// HashForRegion returns a hash of the EC2NodeClass with region-specific filtering applied.
+// This ensures that unsupported features in certain regions don't cause drift.
+func (in *EC2NodeClass) HashForRegion(region string) string {
```

Comment:
What do you think about passing region as an option on Karpenter startup? That way we can just pass the context and the region karpenter is running is available across the project